src/main/cpp/native_glue.hh

```diff
@@ -46,36 +46,26 @@ namespace litecore {
         bool initC4Replicator(JNIEnv *); // Implemented in native_c4replicator.cc
         bool initC4Socket(JNIEnv *);     // Implemented in native_c4socket.cc
 
-        ssize_t UTF8ToModifiedUTF8(const char* input, const char** output, size_t len);
-
         // Creates a temporary slice value from a Java String object
         class jstringSlice {
         public:
             jstringSlice(JNIEnv *env, jstring js);
 
-            ~jstringSlice();
-
             jstringSlice(jstringSlice &&s) // move constructor
-                    : _slice(s._slice), _env(s._env), _jstr(s._jstr) {
-                s._env = nullptr;
+                    : _str(s._str), _slice(s._slice) {
+                s._str = nullptr;
```

Comment:
Use `_str(std::move(s._str))` — that way `string`'s move constructor will be invoked, avoiding copying `_str`.
The 2nd line is unnecessary because `string` already manages its memory safely, so there's no need to clear it.
And the line below is unnecessary because `s`'s destructor doesn't do anything with `_slice` anymore.

Reply:
Neato!  So, the final code looks like this? ```
jstringSlice(jstringSlice &&s): _str(std::move(s._str)), _slice(s._slice)
```

Reply:
Fixed.